[JARS-118] Handle duplicate filenames

hashknot opened the pull request:
Filenames can be repeated while exporting resources
in a dataset. Ensure both the files are present in the
exported dataset zip file.

hashknot commented:
> for some reason, I get a bunch of giles exceptions that the token is invalid (it does download some of them though). This might be a different issue though?
> oh, I just realized that's coming from a different ticket, doesn't it?

That issue was observed when one tried creating a dataset in Amphora without having an active account on Giles. I am not observing this issue locally.

jdamerow commented:
Hm, that can’t be the issue though, since some of the files get downloaded ok. And I have an active account.

---
Julia Damerow, PhD
Scientific Software Engineer
ASU-SFI Center for Biosocial Complex Systems
Digital Innovation Group
Arizona State University
email: jdamerow@asu.edu <mailto:jdamerow@asu.edu> <>
web: http://diging.asu.edu/ <http://diging.asu.edu/> <>







> On Apr 10, 2018, at 11:41 AM, Jitesh <notifications@github.com> wrote:
> 
> for some reason, I get a bunch of giles exceptions that the token is invalid (it does download some of them though). This might be a different issue though?
> oh, I just realized that's coming from a different ticket, doesn't it?
> 
> That issue was observed when one tried creating a dataset in Amphora without having an active account on Giles. I am not observing this issue locally.
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub <https://github.com/diging/amphora/pull/124#issuecomment-380205740>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AIeD9UAgFH5GPrhMXYrxH1b49f7Nn144ks5tnPzdgaJpZM4TKwyi>.
>

hashknot commented:
Didn't you get all the files you requested for in the dataset? Can you give me any pointers on how can I reproduce this issue locally?

jdamerow commented:
Never mind, my fault. Wrong Giles instance… 
---
Julia Damerow, PhD
Scientific Software Engineer
ASU-SFI Center for Biosocial Complex Systems
Digital Innovation Group
Arizona State University
email: jdamerow@asu.edu <mailto:jdamerow@asu.edu> <>
web: http://diging.asu.edu/ <http://diging.asu.edu/> <>







> On Apr 10, 2018, at 11:45 AM, Jitesh <notifications@github.com> wrote:
> 
> Didn't you get all the files you requested for in the dataset? Can you give me any pointers on how can I reproduce this issue locally?
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub <https://github.com/diging/amphora/pull/124#issuecomment-380207056>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AIeD9dhMCfq1s6IO9DA9xOHZeSJT01CBks5tnP3WgaJpZM4TKwyi>.
>

hashknot commented:
- Fixed problems with exported images
- Added support for exporting collections as a whole along with nested sub-collections.
- Now displaying content-type instead of "None" in resource title.

jdamerow commented on cookies/accession/giles.py:
```diff
@@ -56,7 +56,11 @@ def get(self, target, raw=False):
-        response = requests.get(self.sign_uri(target), headers=headers)
+
+        # Giles doesn't return images if 'dw' parameter is missing
+        params = {'dw': 500}
```
we probably want this to be configurable during export, but I'll create a new ticket for that.

jdamerow commented:
one last thing, and then we should be good to go: let's add the collection name and id to which a resource belongs to the index.csv file.

hashknot commented:
Collection ID, name are now included in index.csv file.
